src/main/groovy/com/morpheusdata/hyperv/HyperVCloudProvider.groovy

```diff
@@ -479,21 +519,20 @@ class HyperVCloudProvider implements CloudProvider {
 					continue
 				}
 				if (hypervisor.enabled == true) {
-					def hypervOpts = HypervOptsUtility.getHypervZoneOpts(context, cloud)
-					hypervOpts += HypervOptsUtility.getHypervHypervisorOpts(hypervisor)
-					def hostOnline = ConnectionUtils.testHostConnectivity(hypervOpts.sshHost, 5985, false, true, null)
-					log.debug ("hostOnline: ${hostOnline}")
+					def hypervHypervisorOpts = HypervOptsUtility.getHypervHypervisorOpts(hypervisor)
+					def hostOnline = ConnectionUtils.testHostConnectivity(hypervHypervisorOpts.sshHost, 5985, false, true, null)
+					log.debug("hostOnline: ${hostOnline}")
 					if (hostOnline) {
 						allOnline = hostOnline && allOnline
 						anyOnline = hostOnline || anyOnline
 						//check if this host is online
 						def hostResults = loadHypervHost([zone: cloud], hypervisor)
-						if (hostResults.success == true && hostResults.host && hostResults.host['ComputerName']) {
+						if (hostResults.success == true && hostResults.host && hostResults?.host['ComputerName']) {
 							resolveUniqueIdsToVMids([zone: cloud], hypervisor)
-							def results = listVirtualMachines(hypervOpts)
```

Comment:
listVirtualMachines and related should all be in a VirtualMachineSync service.

Reply:
to test the connection, fetching only one virtual machine